api/src/resources/projects.py

```diff
@@ -73,3 +77,53 @@ def get(self, id):
 
 
 api.add_resource(ProjectLocationsAPI, '/projects/<int:id>/locations', endpoint='project_locations')
+
+
+class UploadAPI(Resource):  
+    def __init__(self):
+        super().__init__()
+        self.parser = reqparse.RequestParser()  # for input validation
+        self.parser.add_argument('file')   # TODO: where should this add be?
+    
+    def post(self):
+        """
+        Test this endpoint using curl or the requests library
+        ex: using curl enter the following in a command prompt:
+        curl -X POST -F file=@"local_filepath" http://localhost:8001/uploads
+
+        check that projects were added to the database by going to the /projects endpoint
+        """
+        # Pull in arguments from the post request
+        file = request.files['file']
+
+        # TODO: add error handling or settings to specify which file extensions are allowed
+        # TODO: add error handling, gives error if trying to add the same project with the same name again. Need to handle that error
+            #sqlite3.IntegrityError: UNIQUE constraint failed: projects.name
```

Comment:
Have you tested it with a postgres database? Having the sqlite3 error is good, but is only relevant for development. We will also want to know what error psycopg2 raises so we can handle it in production.

Reply:
Also, we may not want it to raise errors when multiple projects have the same name. At the moment, there can be multiple records in the projects table with the same project name and different years. If we were to split project details (id, year, ghg, gge, project_id) into a separate table so we can more accurately model the 1 to many relationship between projects and yearly statistics for a project, then we could make project name a unique identifier for the projects table.

A simpler solution might be to change the schema of the projects table so that an error is raised if we try to insert a record that has the same project name and year as an existing record. 

This is definitely more of a big picture question though, and probably does not need to be addressed before this PR can be merged.